### PR TITLE
Full review of map pattern documentation - Issue 164

### PR DESCRIPTION
**Pull request by jdgarciauc3m**
Revised fully the documentation of map in mark-down.

Revised documentation for map API in all headers.

Slightly reworked code.

### CODE REVIEW

**Review by AndresSanchezCuadrado — COMMENTED** [review]
Documentation understandable and changes in the map files allow a correct compilation.

File: doc/map.md [hunk 1]
```diff
@@ -35,15 +47,17 @@ U res = op(x1,x2,...,xN)
...
-The only interface currently offered for this pattern is based in iterators (following the C++ standard library conventions):
+The only interface currently offered for this pattern is based in iterators
+(following the C++ standard library conventions):
 
   * The input data set is specified by two iterators.
   * The output data set is specified by an iterator to the start of the output sequence.
 
 ---
-**Example**
+**Example**: Squaring values in a vector.
```
Comment (Drioastorga) [inline doc/map.md]: The example doubles the values in the vector instead of square them. 

The description or the example should be modified .

**Review by Drioastorga — CHANGES_REQUESTED** [review]
It's necessary to modify the unary map example to match with the description.

**Review by javidoc00 — APPROVED** [review]
OK

**Review by mdolz — APPROVED** [review]
OK

**Review by Drioastorga — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Solves issue #164

**Pull request merged**